spec/controllers/api/orders_controller_spec.rb

```diff
@@ -285,6 +285,18 @@ module Api
           expect(order.reload.pending_payments.empty?).to be true
           expect_order
         end
+
+        context "when payment is not required" do
+          before do
+            allow_any_instance_of(Spree::Order).to receive(:payment_required?) { false }
```

Comment:
I'm not very fond of this and would very much prefer to stub the finder method to return the `order` we care about but it's not a big deal I guess